source/common/upstream/eds.cc

```diff
@@ -108,27 +108,28 @@ void EdsClusterImpl::updateHostsPerLocality(HostSet& host_set, HostVector& new_h
     // If local locality is not defined then skip populating per locality hosts.
     const Locality local_locality(local_info_.node().locality());
     ENVOY_LOG(trace, "Local locality: {}", local_info_.node().locality().DebugString());
-    if (!local_locality.empty()) {
-      std::map<Locality, HostVector> hosts_per_locality;
 
-      for (const HostSharedPtr& host : *current_hosts_copy) {
-        hosts_per_locality[Locality(host->locality())].push_back(host);
-      }
+    std::map<Locality, HostVector> hosts_per_locality;
+
+    for (const HostSharedPtr& host : *current_hosts_copy) {
+      hosts_per_locality[Locality(host->locality())].push_back(host);
+    }
 
-      // Populate per_locality hosts only if upstream cluster has hosts in the same locality.
-      if (hosts_per_locality.find(local_locality) != hosts_per_locality.end()) {
-        per_locality.push_back(hosts_per_locality[local_locality]);
+    const bool local = !local_locality.empty() &&
+                       hosts_per_locality.find(local_locality) != hosts_per_locality.end();
+
+    if (local) {
```

Comment:
There should have been more comments here to begin with. Can you add a comment about how the local locality is first in the vector if it exists?